### PR TITLE
BB2-2191: Terraform BB2 BFD-Insights components part3

### PR DESCRIPTION
**Pull request by dtisza1**
**JIRA Ticket:**
[BB2-2191](https://jira.cms.gov/browse/BB2-2191)

**User Story or Bug Summary:**

Terraforming the **current** configuration of BB2 components for BFD Insights. This is in order to better manage resources and to preserve configuration in the event of a "disaster".

This includes Terraform for the **part 3 (of 3)** related Jira stories. This is split out from the larger DRAFT PR [here](https://github.com/CMSgov/beneficiary-fhir-data/pull/1777).

#### NOTES:

-  Developing using the `allenv` workspace since this is a shared component. We will work to split this out in to `test,impl,prod` environments at a later date and not as part of this work.
- The resources are imported in to the state for the `allenv` workspace currently.

- The Terraform plan is showing the expected changes and they will be applied after PR review is complete!

#### The following is a summary of work done for this PR:

- The `default_tags` under the `bb2/services/analytics` path are:
  - business = "OEDA"
  - application = "bfd-insights"
  - project = "bb2"
  - Environment = "allenv"
  - stack = "bfd-insights-bb2-allenv"

- Terraformed the Update-Tables-for-QuickSight Lambda Function and schedules under `bb2/services/analytics`
  - Resource managed under `modules/event_bridge`:
    - resource "aws_cloudwatch_event_rule" "rule_schedule"
  - Resource managed under `modules/lambda`:
    - resource "aws_lambda_function" "update_athena_metric_tables_lambda"
  - Lambda source code and development files are located under: `modules/lambda/update_athena_metric_tables/*`
    - **NOTE:** This sub-directory is being relocated from the BB2 repository. We have been tracking the code there outside of Terraform for the currently running lambda. The last change was 4-months ago in 3/2023. It contains 9 files.
      - The goal of this PR is to Terraform the currently successfully running lambda; however, feedback related to that actual code is welcomed.
    - See README.md for details.
    - Currently located in the BB2 repository [here](https://github.com/CMSgov/bluebutton-web-server/tree/master/insights/lambdas/update_athena_metric_tables).
      - These files were copied **unchanged** from the BB2 repo. 
      - TODO: This will be removed from the BB2 repository later.
      - For historical context, see [BB2's /insights/lambdas/update_athena_metric_tables/ at 77b9033](https://github.com/CMSgov/bluebutton-web-server/tree/77b90337f0c7ec254d7c80088ab8b3848a3b415e/insights/lambdas/update_athena_metric_tables)

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* The part 2 PR should be merged prior to this. The only conflicting file is the README.md.

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: insights/terraform/projects/bb2/services/analytics/main.tf [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    project     = local.project
+    Environment = local.env
+    stack       = "${local.application}-${local.project}-${local.env}"
+  }
+  application = "bfd-insights"
+  project     = "bb2"
+  env         = terraform.workspace
+  region      = "us-east-1"
+
+  # Shared lambda name/role (TODO: Split out by env)
```
Comment (billie-rose) [inline insights/terraform/projects/bb2/services/analytics/main.tf:14]: Is there a story or ticket we can reference here for this TODO?

File: insights/terraform/projects/bb2/services/analytics/modules/event_bridge/variables.tf [hunk 2]
```diff
@@ -0,0 +1,11 @@
+variable "name" {
```
Comment (billie-rose) [inline insights/terraform/projects/bb2/services/analytics/modules/event_bridge/variables.tf:1]: Can we add descriptions to these variables?

File: insights/terraform/projects/bb2/services/analytics/modules/lambda/main.tf [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+locals {
+  account_id               = data.aws_caller_identity.current.account_id
+  lambda_role_arn          = "arn:aws:iam::${local.account_id}:role/service-role/${var.role}"
+  lambda_function_name_arn = "arn:aws:lambda:${var.region}:${local.account_id}:function:${var.name}"
+  source_dir               = "${path.module}/update_athena_metric_tables/"
+}
+
+
+# Zip File containing Lambda script
+data "archive_file" "zip-archive-update_athena_metric_tables" {
```
Comment (billie-rose) [inline insights/terraform/projects/bb2/services/analytics/modules/lambda/main.tf:16]:
```suggestion
data "archive_file" "zip_archive_update_athena_metric_tables" {
```

File: insights/terraform/projects/bb2/services/analytics/modules/lambda/main.tf [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  description                    = var.description
+  filename                       = data.archive_file.zip-archive-update_athena_metric_tables.output_path
+  handler                        = "lambda_function.lambda_handler"
+  layers                         = []
+  memory_size                    = 128
+  package_type                   = "Zip"
+  reserved_concurrent_executions = -1
+  role                           = local.lambda_role_arn
+  publish                        = false
+  runtime                        = "python3.9"
```
Comment (billie-rose) [inline insights/terraform/projects/bb2/services/analytics/modules/lambda/main.tf:44]: The other lambda python version is `3.8`, just checking this is correct

File: insights/terraform/projects/bb2/services/analytics/modules/lambda/variables.tf [hunk 5]
```diff
@@ -0,0 +1,23 @@
+variable "name" {
```
Comment (billie-rose) [inline insights/terraform/projects/bb2/services/analytics/modules/lambda/variables.tf:1]: Can we add descriptions to these variables?

**Comment by billie-rose** [comment]
Just a few comments. Thank you!

File: insights/terraform/projects/bb2/services/analytics/main.tf (hunk 1 above)
Comment (dtisza1) [reply insights/terraform/projects/bb2/services/analytics/main.tf:14]: @billie-rose I created a backlog story related to this:  https://jira.cms.gov/browse/BB2-2515   It still needs to be groomed and may get split out in to more tasks.  TY!

File: insights/terraform/projects/bb2/services/analytics/modules/lambda/main.tf (hunk 4 above)
Comment (dtisza1) [reply insights/terraform/projects/bb2/services/analytics/modules/lambda/main.tf:44]: @billie-rose This is the version matching the plan.

**Comment by dtisza1** [comment]
> Just a few comments. Thank you!

@billie-rose I just pushed up changes from your review suggestions. Let me know if they look OK? Thank you for reviewing!

**Review by oragame — APPROVED** [review]
Everything looks solid to me. Nice work! local workspace test  on allenv looks good, all up to date.

**Review by billie-rose — APPROVED** [review]

**Pull request merged**